### PR TITLE
Add a stress test under various malloc_conf settings.

### PR DESCRIPTION
**Pull request by guangli-dai**

### CODE REVIEW

File: malloc_conf_tests.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    malloc_configs = [
+        'tcache:false,prof:false',
+        'tcache:false,prof:true',
+        'tcache:true,prof:false,ncached_max:1-1024:100|1025-2048:1|2049-1000000:0,tcache_max:4096',
+        'tcache:true,prof:true,ncached_max:1-1024:100|1025-99999:1|2049-1000000:0,tcache_max:4096'
+    ]
+
+    for config in malloc_configs:
+        run_stress_test(config)
+        if 'ncached_max' in config:
```
Comment (interwq) [inline malloc_conf_tests.py]: Maybe always do the additional tests?

Question: does this add parallelism / do multiple of the test instances run at the same time?

File: malloc_conf_tests.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import subprocess
```
Comment (interwq) [inline malloc_conf_tests.py]: re: name of the test, maybe include the keyword `stress_test` and avoid malloc_conf which is more of a detail here?

File: malloc_conf_tests.py (hunk 1 above)
Comment (guangli-dai) [reply malloc_conf_tests.py]: > Question: does this add parallelism / do multiple of the test instances run at the same time?

No, they are executed sequentially. Do we want parallelism here?

File: malloc_conf_tests.py (hunk 2 above)
Comment (guangli-dai) [reply malloc_conf_tests.py]: Renamed it to `conf_stress_tests.py` mainly to separate it from the `stress_test` folder that already exists and highlight that various confs are the keys here. I am really bad at naming, please suggest if any better ideas.

File: malloc_conf_tests.py (hunk 1 above)
Comment (interwq) [reply malloc_conf_tests.py]: We don't have to make it parallel here. Let's increase the number of threads even higher, maybe 256, given that we have intentional pause / sleep during the benchmark.

File: conf_stress_tests.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            f'--num_threads={num_threads}',
+            f'--num_runs={num_runs}',
+            f'--malloc_size={malloc_size}'
+        ],
+        env=env,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    stdout, stderr = process.communicate()
+    print(f'MALLOC_CONF={malloc_conf}, '
```
Comment (interwq) [inline conf_stress_tests.py]: Let's split this print, first printing out the malloc conf, saying `Running with MALLOC_CONF=...` before invoking the test, then when it's finished output the return value.

This helps if some test is stuck and we want to separately repro using the current conf

**Review by interwq — APPROVED** [review]
@guangli-dai shall we merge this as well?

**Comment by guangli-dai** [comment]
> @guangli-dai shall we merge this as well?

I think we should also add percpu_arena option into the test considering it used to cause unexpected errors. If the addition looks good, I think we are good to merge.

**Comment by interwq** [comment]
>  add percpu_arena option into the test considering it used to cause unexpected errors. 

Good point. We are in no rush merging here. So feel free to include that as well.

**Comment by interwq** [comment]
NVM. Just saw that you already did. Merging.

**Pull request merged**